src/Sushi.Mediakiwi.API/Controllers/AssetController.cs

```diff
@@ -0,0 +1,271 @@
+                temp.Add(new ListItemCollectionOption() 
+                { 
+                    Text = gallery.CompletePath, 
+                    Value = gallery.ID.ToString(), 
+                    IsEnabled = true 
+                });
+            }
+
+            return Ok(temp);
+        }
+
+        /// <summary>
+        /// Saves the supplied asset to Azure and returns information about the created/updated asset
+        /// </summary>
+        /// <param name="request">Contains all information about the asset</param>
+        /// <returns></returns>
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [HttpPost("SaveAsset")]
+        public async Task<ActionResult<SaveAssetResponse>> SaveAsset([FromForm] SaveAssetRequest request)
+        {
+            // Request is empty
+            if (request == null)
+            {
+                return BadRequest();
+            }
+
+            // New image
+            if (request.ID.GetValueOrDefault(0) == 0)
+            {
+                // But no data supplied
+                if (request.Data?.Length == 0 || string.IsNullOrWhiteSpace(request.Data?.FileName) == true)
+                {
+                    return BadRequest();
+                }
+
+                // But no gallery supplied
+                if (request.GalleryID.GetValueOrDefault(0) == 0)
+                {
+                    return BadRequest();
+                }
+            }
+
+            // The target asset
+            Data.Asset targetAsset = new Data.Asset();
+
+            // We're updating an asset, get it from DB
+            if (request.ID.GetValueOrDefault(0) > 0)
+            {
+                targetAsset = await Data.Asset.SelectOneAsync(request.ID.Value);
+            }
+
+            // We have new data provided, process it
+            if (request.Data != null)
+            {
+                targetAsset.Type = request.Data.ContentType;
+                targetAsset.Extention = request.Data.FileName.Substring(request.Data.FileName.LastIndexOf('.'));
+                targetAsset.FileName = request.Data.FileName;
+                targetAsset.Size = request.Data.Length;
+
+                // set the new filename
+                targetAsset.FileName = request.Data.FileName;
+
+                // Whenever it's an image, extract image data
+                if (request.Data.ContentType.Equals("image/jpeg", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/jpg", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/gif", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/png", System.StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Is there any special handling of BMPs?

Reply:
I added the image/bmp contenttype, though i hope no one in their right mind uploads a BMP 😃